spec/npm/npmSpec.js

```diff
@@ -0,0 +1,52 @@
+const fs = require('fs');
+
+describe('The npm module', function() {
+	let testEnv;
+
+	beforeEach(function() {
+		testEnv = this.getTestEnv();
+	});
+
+	it('should install and run simply named packages', async function() {
+		const scriptSource = `
+			const uniqueCount = npm.dedupe([1, 2, 2, 3]).length;
+			here.file('output').make(true).content = uniqueCount;
+		`;
+
+		await testEnv.runLemonScript(scriptSource);
+
+		const outputPath = testEnv.nativePathFor('output');
+		let value = parseInt(fs.readFileSync(outputPath));
+		expect(value).toBe(3);
+	});
+
+	it('should install and run scoped packages', async function() {
+		const scriptSource = `
+			const octokit = new npm['@octokit/rest'].Octokit();
+
+			let lemon = (await octokit.rest.repos.listForUser({ username: "Cykelero" }))
```

Comment:
Accessing the network is inevitable to test npm support, but it'd be good to avoid making an extra network call when checking the package itself, as with `listForUser` here. Do you think you could find a scoped package that doesn't require network access?

Reply:
Probably. Does this test add any value that isn't covered by `should recognize pinned versions for scoped packages`? When writing, I thought of the pinned versioning after the main fix and added a test for it, but looking at it now, it seems like it also checks the scoped package part. Maybe drop down to just the one test?

Reply:
I think you were right to create the two tests; this one is less specific, and it might not fail when `recognize pinned versions`, helping narrow down potential issues.

And, thinking of it—since all these tests don't reset the package bundle cache, if all goes well they shouldn't access the network at all, when making use of `npm`, once the cache has been primed. Meaning, if you manage to replace the `listForUser` call with something that doesn't hit the network, the tests will add virtually no time to test runs!